api/candlepin-api-spec.yaml

```diff
@@ -1343,6 +1517,90 @@ components:
             upstreamConsumer:
               $ref: '#/components/schemas/UpstreamConsumerDTO'
 
+    ProductContentDTO:
+      description: Product content mapping exposed to the API
+      required:
+        - id
+      properties:
+        content:
+          $ref: "#/components/schemas/ContentDTO"
+        enabled:
+          type: boolean
+
+    ProductLegacyDTO:
+      description: DTO representing the product data exposed to the API
+      required:
+        - id
+      allOf:
+        - $ref: '#/components/schemas/TimestampedEntity'
+        - type: object
+          properties:
+            id:
+              type: string
+              example: 5051
+            uuid:
+              type: string
+              example: ff808081554a3e4101554a3e9033005d
+            name:
+              type: string
+              example: Admin OS Developer Bits
+            multiplier:
+              type: integer
+              format: int64
+              example: 1
+            attributes:
+              type: array
+              items:
+                type: object
+                additionalProperties:
+                  type: string
+            productContent:
+              type: array
+              uniqueItems: true
+              items:
+                $ref: "#/components/schemas/ProductContentDTO"
+            dependentProductIds:
+              type: array
+              uniqueItems: true
+              items:
+                type: string
+            branding:
+              type: array
+              uniqueItems: true
+              items:
+                $ref: "#/components/schemas/BrandingDTO"
+            href:
+              type: string
+              example: /products/ff808081554a3e4101554a3e9033005d
+            locked:
+              type: boolean
+
+    ProductDTO:
```

Comment:
I am very confused here. I assume this is the main ProductDTO, and that the ProductLegacyDTO is supposed to be the DTO for ProductData. If this is supposed to be the original ProductDTO, then it definitly should not be inheriting from the ProductLegacyDTO, and ProductLegacyDTO should definitely NOT be used throughout our codebase as a replacement for ProductDTO. ProductLegacyDTO should be used only for the member fields on the SubscriptionDTO (since this is a legacy dto structure and should go away eventually).

I feel we should not be blending/inheriting these two. We should do either of:
-  There should be a ProductDTO specified fully based on what our current ProductDTO looks like, and then a ProductLegacyDTO that is separate and fully based on what ProductData look like now. 
- If we chose to do inheritance (even though I think we shouldn't), then it would have to go the other way: Define ProductDTO based on what the current ProductDTO looks like, and then ProductLegacyDTO can inherit from that (if needed).
- Ideally, we could specify ONLY ProductDTO, which will be used for both purposes, since the structure is almost the same for our current ProductDTO and ProductData.

@Januson @Ceiu what do you think?

Also I'm not sure what's happening with the AttributesDTO, why is it added here, while there's already an attributes field on the legacyDTO.

Reply:
Done. *LegacyDTO dtos were removed